### DISCUSSION TITLE
Idle question about the battery camera situation

### DISCUSSION THREAD

**Discussion by f3Z0zRM** [QuantumEntangledAndy/neolink#174] [General]
Sorry if this isn't a good topic for this forum. I'm just hoping the people closest to the real facts while not being dead set against talking about them are here.

I'm an argus 2e owner and eventually noticed that I couldn't login to the camera unless my internet connection was available and then started googling. I found mostly stuff affirming that Reolink battery cameras did not support local login. Contradicting that was "[Whether Reolink Battery-powered Cameras Can Work without WiFi or Internet Connection](https://support.reolink.com/hc/en-us/articles/360006943713-Whether-Reolink-Battery-powered-Cameras-Can-Work-without-WiFi-or-Internet-Connection/)" on Reolink's own website and I also did try Reolink tech support about the topic and just got what I interpret as gaslighting that the linked support article was entirely accurate and relevant to users of the Reolink client though conceivably they were just trained wrong or something.

Eventually I noticed that this branch of Neolink could actually do local login to my argus 2e, which surprised me greatly since I figured Reolink just hadn't ever really aimed for that.

So now I'm very curious whether there's a plausible explanation for why Reolink might be playing it this way: not implementing local login for battery cameras in the client, saying they did, for years on end, all while the battery camera hardware apparently does support local login? Alternatively, am I mistaken about this narrative?

**Comment by QuantumEntangledAndy** [comment]
From what I have seen of the network packets that the official client sends out it does do local login. The login sequence for my Argus 2E involves all five login methods at once. Indeed the protocol in neolink is based on what the client does with my 2E.

However local login only works in the following situations:

- the router supports broadcast

Some router will block broadcasts. So this method dosent always work.

- you know both the IP address and the uid

Reolink get the IP address from its own servers as the camera will phone home to register the IP address

The uid is needed for the hello call to the camera once the ip is know. 

---

What neolink does when it knows both ip and uid is initiate the same connection that a broadcast does but instead of using the usually blocked broadcast packet it sends out the same packet contents but directly to the ip (which won't be blocked by the router).

So if your network supports broadcasts everything works fine with the official client. If not Reolink will only get the ip from phoning home.

In neolink you have one additional option to send the broadcast packet straight to an ip
